flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/dialect/AbstractPostgresCompatibleDialect.java

```diff
@@ -50,12 +50,14 @@ public String getLimitClause(long limit) {
     @Override
     public Optional<String> getUpsertStatement(
             String tableName, String[] fieldNames, String[] uniqueKeyFields) {
+        Set<String> uniqueKeyFieldsSet = Arrays.stream(uniqueKeyFields).collect(Collectors.toSet());
         String uniqueColumns =
-                Arrays.stream(uniqueKeyFields)
+                uniqueKeyFieldsSet.stream()
                         .map(this::quoteIdentifier)
                         .collect(Collectors.joining(", "));
         String updateClause =
                 Arrays.stream(fieldNames)
+                        .filter(f -> !uniqueKeyFieldsSet.contains(f))
```

Comment:
I'm curious is this a bug or performance improvement?

Also cc @RocMarshal , would you mind reviewing this PR as well (as you are the active contributor of jdbc connector)?

Reply:
It's better to add a test to check it if it's a bug.

Reply:
Thanks @Mrart for the contribution & @1996fanrui for the ping.

> It's better to add a test to check it if it's a bug.

+1.